Mathlib/GroupTheory/Perm/Cycle/Basic.lean

```diff
@@ -1510,21 +1510,114 @@ theorem cycle_is_cycleOf {f c : Equiv.Perm α} {a : α} (ha : a ∈ c.support)
   suffices f.cycleOf a = c.cycleOf a by
     rw [this]
     apply symm
-    exact
-      Equiv.Perm.IsCycle.cycleOf_eq (Equiv.Perm.mem_cycleFactorsFinset_iff.mp hc).left
-        (Equiv.Perm.mem_support.mp ha)
-  let hfc := (Equiv.Perm.disjoint_mul_inv_of_mem_cycleFactorsFinset hc).symm
-  let hfc2 := Perm.Disjoint.commute hfc
-  rw [← Equiv.Perm.cycleOf_mul_of_apply_right_eq_self hfc2]
+    exact IsCycle.cycleOf_eq (mem_cycleFactorsFinset_iff.mp hc).left (mem_support.mp ha)
+  let hfc := (disjoint_mul_inv_of_mem_cycleFactorsFinset hc).symm
+  let hfc2 := Disjoint.commute hfc
+  rw [← cycleOf_mul_of_apply_right_eq_self hfc2]
   simp only [hfc2.eq, inv_mul_cancel_right]
-  -- a est dans le support de c, donc pas dans celui de g c⁻¹
-  exact
-    Equiv.Perm.not_mem_support.mp
-      (Finset.disjoint_left.mp (Equiv.Perm.Disjoint.disjoint_support hfc) ha)
+  -- a belongs to the support of c, hence not to that of g * c⁻¹
+  exact not_mem_support.mp (Finset.disjoint_left.mp (Disjoint.disjoint_support hfc) ha)
 #align equiv.perm.cycle_is_cycle_of Equiv.Perm.cycle_is_cycleOf
 
+theorem eq_cycleOf_of_mem_cycleFactorsFinset_iff
+    (g c : Perm α) (hc : c ∈ g.cycleFactorsFinset) (x : α) :
+    c = g.cycleOf x ↔ x ∈ c.support := by
+  refine ⟨?_, (cycle_is_cycleOf · hc)⟩
+  rintro rfl
+  rw [mem_support, cycleOf_apply_self, Ne.def, ← cycleOf_eq_one_iff]
+  exact (mem_cycleFactorsFinset_iff.mp hc).left.ne_one
+
+/-- A permutation `c` is a cycle of `g` iff `k * c * k⁻¹` is a cycle of `k * g * k⁻¹` -/
```

Comment:
The sense of the double implication is swapped in the docstring wrt to the actual statement. Also, I would call it `mem_cycleFactorsFinset_iff`.

Reply:
I changed the docstring and the name.